### PR TITLE
Add publish-new-version.sh for pushing versions of the crate

### PR DESCRIPTION
**Pull request by acmcarther**
Adds a script for publishing new versions of this crate to codify my shell history into a shell script usable by other folks.

Has some rudimentary checks for the existence of `cargo` and that the working directory is clean.

r: @UebelAndre as well

### CODE REVIEW

**Comment by UebelAndre** [comment]
@acmcarther Are there also some examples of using this script? Whenever I add things like this to projects at work, I typically add a block of comments at the top of the file that show simple usage just to make it **crystal clear** as to what to do.
```bash
# Use the script like so
# Args: ...docs...
#
# ./my_script.sh arg1 arg2
```
```bash
# Use the script like so
# Environment: ...docs...
#
# ARG_VAR=1 ./my_other_script.sh
```
Things like that. 

While It's unlikely I'll be doing a deploy for this so If there's some internal Google doc that has this, that works as well. The thing I want to avoid is all maintainers who have familiarity with the project/script being unavailable and therefore no more releases happen because maybe some constraint changed or things are not as obvious to the available/new maintainers. 

Whether or not you decide to do that, I'm thankful for this PR! 😄

**Comment by acmcarther** [comment]
Thanks, that's valuable feedback.

I've added some documentation about what the script does and how to use it to the top of the file. Incidentally, I've also updated the mechanism it uses to take in the new version from an environment variable to a regular argument.

File: publish-new-version.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    command -v "$1" >/dev/null 2>&1 || ( echo "Command \`$1\` isn't available. Please install before continuing."; exit 1 )
+}
+
+update_cargo_toml_version() {
+  crate_version=$1
+  sed -i "s/^version =.*/version = \"$crate_version\"/g" Cargo.toml
+}
+
+cargo_build_crate() {
+  cargo build
```
Comment (UebelAndre) [inline publish-new-version.sh:28]: Should this have `--release` with it?

File: publish-new-version.sh [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+if ! [ -z "$(git status --porcelain)" ]; then
+  echo "Working directory is not clean. Commit pending changes before running command."
+  exit 1
+fi
+
+PWD="$(pwd)"
+
+REPO_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd "$REPO_ROOT/impl"
```
Comment (UebelAndre) [inline publish-new-version.sh:83]: How would you feel about `set -x` here? Might make logging of what the deploy process is doing nicer in the event of something going wrong.

**Review by damienmg — APPROVED** [review]

**Review by UebelAndre — APPROVED** [review]

File: publish-new-version.sh (hunk 2 above)
Comment (acmcarther) [reply publish-new-version.sh:83]: Now done!

File: publish-new-version.sh (hunk 1 above)
Comment (acmcarther) [reply publish-new-version.sh]: Done.

Added explanation for why we're doing this as well.

**Review by UebelAndre — APPROVED** [review]

**Comment by UebelAndre** [comment]
@acmcarther When the version is updated should the [`Cargo.lock`](https://github.com/google/cargo-raze/blob/master/tools/Cargo.lock) file be updated too in the `/tools` directory?

**Comment by acmcarther** [comment]
Re https://github.com/google/cargo-raze/pull/217#issuecomment-694394354
 
I'm not really sure. I don't consider the contents of `tools/` to be part of the package proper though, so i think it's OK to punt on this question for now.

That said, this should be a consideration in handling https://github.com/google/cargo-raze/issues/210 I think.

**Comment by UebelAndre** [comment]
As a quick followup. Do you think you could delete some old branches? 😅

**Comment by acmcarther** [comment]
Done!

**Pull request merged**